src/EnergyPlus/DaylightingManager.cc

```diff
@@ -3271,27 +3125,28 @@ void FigureDayltgCoeffsAtPointsForSunPosition(
                         for (int IntWinDisk = thisSpace.WindowSurfaceFirst, IntWinDisk_end = thisSpace.WindowSurfaceLast;
                              IntWinDisk <= IntWinDisk_end;
                              ++IntWinDisk) {
-                            if (state.dataSurface->Surface(IntWinDisk).ExtBoundCond >= 1) {
-                                if (state.dataSurface->Surface(state.dataSurface->Surface(IntWinDisk).ExtBoundCond).Zone ==
-                                    state.dataSurface->Surface(IWin2).Zone) {
-                                    PierceSurface(state, IntWinDisk, RREF, RAYCOS, HitPtIntWinDisk, hitIntWinDisk);
-                                    if (hitIntWinDisk) {
-                                        IntWinDiskHitNum = IntWinDisk;
-                                        COSBIntWin = dot(state.dataSurface->Surface(IntWinDisk).OutNormVec, RAYCOS);
-                                        if (COSBIntWin <= 0.0) {
-                                            hitIntWinDisk = false;
-                                            IntWinDiskHitNum = 0;
-                                            continue;
-                                        }
-                                        TVISIntWinDisk = General::POLYF(
-                                            COSBIntWin,
-                                            state.dataConstruction->Construct(state.dataSurface->Surface(IntWinDisk).Construction).TransVisBeamCoef);
-                                        break;
-                                    }
-                                }
+                            auto const &surfIntWinDisk = state.dataSurface->Surface(IntWinDisk);
+                            if (surfIntWinDisk.ExtBoundCond < 1)
+                                continue;
+                                
+                            if (state.dataSurface->Surface(surfIntWinDisk.ExtBoundCond).Zone != state.dataSurface->Surface(IWin2).Zone)
+                                continue;
+                                
+                            hitIntWinDisk = PierceSurface(state, IntWinDisk, RREF, RAYCOS, HitPtIntWinDisk);
```

Comment:
`PierceSurface` and its variants now return `hit` as opposed to taking a reference to `hit` as a parameter. Return values, people. That's what they're here for.